listenbrainz/db/lb_radio_artist.py

```diff
@@ -0,0 +1,151 @@
+                            LEFT JOIN similarity.overhyped_artists oa
+                                   ON sa.similar_artist_mbid = oa.artist_mbid::TEXT
+                             ORDER BY rownum
+                                LIMIT %s
+                           ), select_similar_artists AS (
+                               SELECT similar_artist_mbid
+                                    , score
+                                 FROM knockdown
+                                WHERE rownum in %s
+                                ORDER BY rownum
+                           ), similar_artists_and_orig_artist AS (
+                               SELECT *
+                                 FROM select_similar_artists
+                                UNION
+                               SELECT *
+                                 FROM mbids
+                           ), combine_similarity AS (
+                               SELECT similar_artist_mbid
+                                    , artist_mbid
+                                    , recording_mbid
+                                    , total_listen_count
+                                    , total_user_count
+                                 FROM popularity.top_recording tr
+                                 JOIN similar_artists_and_orig_artist sao
+                                   ON tr.artist_mbid = sao.similar_artist_mbid::UUID
+                                UNION ALL
+                               SELECT similar_artist_mbid
+                                    , artist_mbid
+                                    , recording_mbid
+                                    , total_listen_count
+                                    , total_user_count
+                                 FROM popularity.mlhd_top_recording tmr
+                                 JOIN similar_artists_and_orig_artist sao2
+                                   ON tmr.artist_mbid = sao2.similar_artist_mbid::UUID
+                           ), group_similarity AS (
+                               SELECT similar_artist_mbid
+                                    , artist_mbid
+                                    , recording_mbid
+                                    , SUM(total_listen_count) AS total_listen_count
+                                    , SUM(total_user_count) AS total_user_count
+                                 FROM combine_similarity
+                             GROUP BY recording_mbid, artist_mbid, similar_artist_mbid
+                           ), top_recordings AS (
+                               SELECT sa.similar_artist_mbid
+                                    , gs.recording_mbid
+                                    , total_listen_count
+                                    , PERCENT_RANK() OVER (PARTITION BY sa.similar_artist_mbid
+                                                               ORDER BY sa.similar_artist_mbid, total_listen_count ) AS rank
```

Comment:
`sa.similar_artist_mbid` is redundant in the `ORDER BY` because you are partitioning by it, so the value will be same inside a partition and the order by in a window function only performs ordering per partition.